src/header/common/content_disposition.rs

```diff
@@ -174,112 +454,423 @@ impl Header for ContentDisposition {
     }
 }
 
-impl fmt::Display for ContentDisposition {
+impl fmt::Display for DispositionType {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self.disposition {
-            DispositionType::Inline => try!(write!(f, "inline")),
-            DispositionType::Attachment => try!(write!(f, "attachment")),
-            DispositionType::Ext(ref s) => try!(write!(f, "{}", s)),
+        match self {
+            DispositionType::Inline => write!(f, "inline"),
+            DispositionType::Attachment => write!(f, "attachment"),
+            DispositionType::FormData => write!(f, "form-data"),
+            DispositionType::Ext(ref s) => write!(f, "{}", s),
         }
-        for param in &self.parameters {
-            match *param {
-                DispositionParam::Filename(ref charset, ref opt_lang, ref bytes) => {
-                    let mut use_simple_format: bool = false;
-                    if opt_lang.is_none() {
-                        if let Charset::Ext(ref ext) = *charset {
-                            if ext.eq_ignore_ascii_case("utf-8") {
-                                use_simple_format = true;
-                            }
-                        }
-                    }
-                    if use_simple_format {
-                        use std::str;
-                        try!(write!(f, "; filename=\"{}\"",
-                                    match str::from_utf8(bytes) {
-                                        Ok(s) => s,
-                                        Err(_) => return Err(fmt::Error),
-                                    }));
-                    } else {
-                        try!(write!(f, "; filename*={}'", charset));
-                        if let Some(ref lang) = *opt_lang {
-                            try!(write!(f, "{}", lang));
-                        };
-                        try!(write!(f, "'"));
-                        try!(header::http_percent_encode(f, bytes))
-                    }
-                },
-                DispositionParam::Ext(ref k, ref v) => try!(write!(f, "; {}=\"{}\"", k, v)),
+    }
+}
+
+impl fmt::Display for DispositionParam {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // TODO: More special charaters in filename should be escaped.
```

Comment:
Btw we already use regex for example, so if you think we'd need some more characters you could add it here too using regex crate I guess.